Prog/Demographic trends.sas

```diff
@@ -0,0 +1,1458 @@
+by upuma;
+	%assign_jurisdiction; 
+run;
+
+proc summary data= vacantunits;
+	class Jurisdiction;
+	var totalunits vacunit;
+	output out = vacancyrate  sum=;
+	format Jurisdiction Jurisdiction. ;
+	run;
+
+data vacancyrate2;  
+set vacancyrate;
+keep Jurisdiction vacrate;
+vacrate= vacunit/totalunits;
+run;
+
+proc sort data=COGSarea_2017 ;
+by Jurisdiction;
+run;
+
+data hh_housing;
+merge COGSarea_2017 vacancyrate2;
+by Jurisdiction;
+run;
+
+data hhestimates17 (drop= _FREQ_);
+set hh_housing;
+hhestimate17= totalunits*(1-vacrate);
+run;
+/*end of estimating 2017 hh counts*/
+
+proc summary data=hhestimates17;
+	class Jurisdiction ;
+	var hhestimate17;
+	weight hhwt;
+	output out = HH17 sum=;
+	format Jurisdiction Jurisdiction. ;
+run;
+
+proc sort data=hhestimates17;
+by Jurisdiction;
+run;
+
+data HH17;
+set HH17;
+if _TYPE_=0 then Jurisdiction=11;
+run;
+proc sort data=population;
+by Jurisdiction;
+run;
+proc summary data=population;
+	class Jurisdiction ;
+	var numhhs9 numhhs0 numhhs1;
+	output out = NCDBhouseholds(drop= _FREQ_)  sum=;
+	format Jurisdiction Jurisdiction. ;
+run;
+
+proc summary data= pop78;
+class Jurisdiction ;
+var tothh70 tothh80;
+output out= households78 sum=;
+format Jurisdiction Jurisdiction. ;
+run;
+data households78;
+set households78;
+if _TYPE_=0 then Jurisdiction=11;;
+run;
+
+data NCDBhouseholds;
+set NCDBhouseholds;
+if _TYPE_=0 then Jurisdiction=11;
+run;
+
+proc sort data=HH17;
+by Jurisdiction;
+run;
+
+proc sort data=households78;
+by Jurisdiction;
+run;
+
+proc sort data=NCDBhouseholds;
+by Jurisdiction;
+run;
+
+data totalhouseholdtrend (drop= _TYPE_ _FREQ_);
+merge households78 NCDBhouseholds HH17;
+by Jurisdiction;
+label NUMHHS9= "Total households in 1990";
+label NUMHHS0= "Total households in 2000";
+label NUMHHS1= "Total households in 2010";
+label hhestimate17 = "Total households in 2017";
+run;
+
+proc export data = totalhouseholdtrend
+   outfile="&_dcdata_default_path\RegHsg\Prog\totalhouseholdtrend.csv"
+   dbms=csv
+   replace;
+run;
+
+/*hh by size, type and income*/
+%macro hhnonrelate(year);
+data hhrelate_&year. (where=(upuma in ("1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002","2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107","5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305","5159306", "5159307", "5159308", "5159309", "5110701", "5110702" , "5110703", "5151244","5151245", "5151246", "5151255"))) ;
+set Ipums.ACS_&year._dc(where=(gq in (1,2))) Ipums.ACS_&year._va(where=(gq in (1,2))) Ipums.ACS_&year._md(where=(gq in (1,2)));
+keep upuma pernum gq hhwt perwt year serial numprec relate related notnonrelate totnumpop;
+if relate in (11,12,13) then notnonrelate=0;
+else if relate in (2,3,4,5,6,7,8,9,10) then notnonrelate=1;
+totnumpop=1;
+run;
+
+proc summary data=hhrelate_&year.;
+  class serial;
```

Comment:
If you use a **class** statement instead of a **by** statement here, put an *nway* option on the proc summary statement. Otherwise, you end up with a total line for all of the people in the data set.